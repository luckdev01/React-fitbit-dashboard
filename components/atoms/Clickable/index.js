import PropTypes from 'prop-types'
import Color from 'color'

import spacing from '../../../styles/spacing'
import colors from '../../../styles/colors'

export default function Clickable ({ children, ...props }) {
  return (
    <a {...props}>
      {children}
      <style jsx>{`
        a {
          display: inline-block;
          padding: ${spacing.medium} ${spacing.xlarge};
          background: ${colors.green};
          border-bottom: 3px solid ${Color(colors.green).darken(0.25)};
          text-decoration: none;
          text-transform: uppercase;
          border-radius: 3px;
          color: ${colors.white};
          font-weight: bold;
          -webkit-font-smoothing: antialiased;
        }
      `}</style>
    </a>
  )
}

Clickable.propTypes = {
  children: PropTypes.any,
}
