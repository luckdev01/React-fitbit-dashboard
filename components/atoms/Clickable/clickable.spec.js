import Clickable from './'

describe('Clickable', () => {
  it('renders children', () => {
    const child = 'this is a test'
    const wrapper = shallow(<Clickable>{child}</Clickable>)
    expect(wrapper).toMatchSnapshot()
  })
})
