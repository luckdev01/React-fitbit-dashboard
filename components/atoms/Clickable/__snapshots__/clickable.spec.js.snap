// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clickable renders children 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Clickable>
    this is a test
  </Clickable>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "this is a test",
        <JSXStyle
          css="a.jsx-1068520229{display:inline-block;padding:12px 24px;background:#46cba0;border-bottom:3px solid hsl(160.60000000000002, 56.1%, 40.1%);text-decoration:none;text-transform:uppercase;border-radius:3px;color:#FFFFFF;font-weight:bold;-webkit-font-smoothing:antialiased;}"
          styleId="1068520229"
        />,
      ],
      "className": "jsx-1068520229 ",
    },
    "ref": null,
    "rendered": Array [
      "this is a test",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "css": "a.jsx-1068520229{display:inline-block;padding:12px 24px;background:#46cba0;border-bottom:3px solid hsl(160.60000000000002, 56.1%, 40.1%);text-decoration:none;text-transform:uppercase;border-radius:3px;color:#FFFFFF;font-weight:bold;-webkit-font-smoothing:antialiased;}",
          "styleId": "1068520229",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "this is a test",
          <JSXStyle
            css="a.jsx-1068520229{display:inline-block;padding:12px 24px;background:#46cba0;border-bottom:3px solid hsl(160.60000000000002, 56.1%, 40.1%);text-decoration:none;text-transform:uppercase;border-radius:3px;color:#FFFFFF;font-weight:bold;-webkit-font-smoothing:antialiased;}"
            styleId="1068520229"
          />,
        ],
        "className": "jsx-1068520229 ",
      },
      "ref": null,
      "rendered": Array [
        "this is a test",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "css": "a.jsx-1068520229{display:inline-block;padding:12px 24px;background:#46cba0;border-bottom:3px solid hsl(160.60000000000002, 56.1%, 40.1%);text-decoration:none;text-transform:uppercase;border-radius:3px;color:#FFFFFF;font-weight:bold;-webkit-font-smoothing:antialiased;}",
            "styleId": "1068520229",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
